common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -111,23 +112,19 @@ def may_certify_for_course(
             course's certificates before the course has ended.
         has_ended (bool): Whether the course has ended.
         certificate_available_date (datetime): the date the certificate is available on for the course.
+        self_paced (bool): Whether the course is self-paced.
     """
     show_early = (
         certificates_display_behavior in ('early_with_info', 'early_no_info')
         or certificates_show_before_end
     )
-    past_availability_date = (
+    past_available_date = (
         certificate_available_date
         and certificate_available_date < datetime.now(utc)
     )
+    ended_without_available_date = (certificate_available_date is None) and has_ended
```

Comment:
I still don't really understand why the available date isn't actually automatically set to the default sometimes...

Reply:
Is that something worth looking into or is that maybe out of the scope of this PR?

Reply:
Hmm, good question.  I'm actually not sure under which circumstances it's not set to the default now.  I'll take a look.  In any case, I don't think there's much harm in being a little defensive here.

Reply:
It should be set when a new object is made, I thought, but for some reason it isn't. Better to be defensive than have it fail, though, right?